components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/deployment/AppDeployerServiceComponent.java

```diff
@@ -210,4 +225,40 @@ public void registerEventSinkDeployer(ArtifactDeploymentManager artifactDeployme
             log.error("Error instantiating EventSinkDeployer class", e);
         }
     }
+
+    /**
+     * Initialize and add the CappDeployer to the Deployment Engine.
+     */
+    private void addCAppDeployer() {
+        try {
+            String appsRepo = "carbonapps";
+            // Initialize CApp deployer here
+            Class deployerClass = Class.
+                    forName("org.wso2.micro.integrator.initializer.deployment.application.deployer.CappDeployer");
+            Deployer deployer = (Deployer) deployerClass.newInstance();
```

Comment:
Any reason why we have 'Deployer deployer = new CappDeployer()' here?

Reply:
CappDeployer is the class that is responsible for deploying carbon applications. I.e. the deploy() method in the CappDeployer will be invoked through the DeploymentEngine class when there are pending carbon applications to be deployed. Therefore, this deployer (and other remaining deployers) should be initialized and registered during the server startup time. 

CAppDeploymentManager class will manage carbon application deployment by maintaining all successfully deployed capps, faulty capps, app deployment handlers such as DefaultAppDeployer, SynapseAppDeployer, etc and by having relevant functionalities for complex capp deployement. 

Also, CAppDeploymentManager.deploy() method will have the actual logic of the capp deployment and will be invoked from CappDeployer.deploy() method.

This separation will help to reduce the complexity of the CappDeployer class and will maintain the common structure as in other deployer classes.

Reply:
Changed to Deployer deployer = new CappDeployer();